saw-script/src/SAWScript/REPL/Monad.hs

```diff
@@ -87,24 +88,22 @@ import SAWScript.ValueOps (makeCheckpoint, restoreCheckpoint)
 
 -- REPL Environment.
 data REPLState = REPLState
-  { rContinue   :: IORef Bool
+  { rContinue   :: Bool
   , rIsBatch    :: Bool
   , rTopLevelRO :: TopLevelRO
-  , rTopLevelRW :: IORef TopLevelRW
-  , rProofState :: Maybe (IORef ProofState)
+  , rTopLevelRW :: TopLevelRW
+  , rProofState :: Maybe ProofState
```

Comment:
It might be worth documenting under what conditions one can expect this field to be `Just`. Some of the panics introduced elsewhere in this comment talk about a "`ProofScript` mode", which seems relevant here.